src/opm/parser/eclipse/EclipseState/Schedule/Group/GuideRate.cpp

```diff
@@ -225,12 +223,13 @@ void GuideRate::compute(const std::string& wgname,
     this->injection_group_values[std::make_pair(phase, wgname)] = group.guide_rate;
 }
 
-void GuideRate::well_compute(const std::string& wgname,
-                             size_t             report_step,
-                             double             sim_time,
-                             double             oil_pot,
-                             double             gas_pot,
-                             double             wat_pot)
+void GuideRate::well_compute(const std::string &wgname,
+                             size_t report_step,
+                             double sim_time,
+                             const bool time_to_update,
```

Comment:
The `GuideRate:.timeToUpdate()` can be called from inside the `GuideRate::well_compute()` and `GuideRate::group_compute()`- no reason to call on the outside and pass in parameter.

Reply:
Yes. The only motivation not putting `timeToUpdate()` inside `well_compute` and `group_compute` is that we do not need to call the function `timeToUpdate()` again and again. In function `timeToUpdate()`, we will potentially go through all the wells and groups.    If there is no concern regarding the performance impact from this,  I can make `timeToUpdate` private and called through `well_compute` and `group_compute`. Please suggest.

Reply:
Ok - I understand; I still think it should go in as suggested.

Reply:
Actually I just found we can not do that. Let us discuss through an example scenario. 

There are four wells, `P1`, `P2`, `P3`, and `P4`.    Based on the `time_delay`, the well `P1` should be updated with the guide rate. `P2`, `P3` and `P4` not yet.  Based on the theory we are trying to enforce through this PR, all the four wells should update their guide rate. 

If we call the function `timeToUpdate` when update `P1`, it returns `true`, then we update `P1`'s guide rate and `P1` gets a new update time. 
Then if we call the function `timeToUpdate` when updating`P2`, because `P1` is already updated, it does not need to update anymore. Then the function `timeToUpdate` returns `false`. Then `P2` and `P3` and `P4` will NOT be updated in this time step.  

Basically I mean, because `well_update` and `group_update` will change the content of `GuideRate`, it makes `timeToUpdate` might return different results. 

Does it make sense to you?